src/runner.ts

```diff
@@ -0,0 +1,578 @@
+          Array.from(this._workspace.repos.entries())
+              .filter(repo => repo[1].test)
+              .sort((a, b) => a[0] < b[0] ? -1 : (a[0] > b[0] ? 1 : 0));
+      console.log(`Repos to test: ${workspaceReposToTest.length}`);
+      for (const repo of workspaceReposToTest) {
+        console.log(`    ${git.serializeGitHubRepoRef(repo[1].githubRepoRef)}`);
+      }
+      const workspaceReposToRequire =
+          Array.from(this._workspace.repos.entries())
+              .filter(repo => !repo[1].test)
+              .sort((a, b) => a[0] < b[0] ? -1 : (a[0] > b[0] ? 1 : 0));
+      if (workspaceReposToRequire.length > 0) {
+        console.log(`Repos to provide, but not test: ${workspaceReposToRequire
+                        .length}`);
+        for (const repo of workspaceReposToRequire) {
+          console.log(
+              `   ${git.serializeGitHubRepoRef(repo[1].githubRepoRef)}`);
+        }
+      }
+    }
+  }
+
+  /**
+   * Given a collection of GitHubRepoRefs, replace any that represent wildcard
+   * values with the literal values after comparing against names of repos on
+   * GitHub.  So a repo ref like `Polymer/*` return everything owned by
+   * Polymer where `PolymerElements/iron-*` would be all repos that start with
+   * `iron-` owned by `PolymerElements` org.
+   */
+  async _expandWildcardRepoRefs(repoRefs: git.GitHubRepoRef[]):
+      Promise<git.GitHubRepoRef[]> {
+    const ownersToFetchRepoNamesFor: Set<string> = new Set();
+    for (const repo of repoRefs) {
+      if (repo.repoName.match(/\*/)) {
+        ownersToFetchRepoNamesFor.add(repo.ownerName.toLowerCase());
+      }
+    }
+    if (ownersToFetchRepoNamesFor.size === 0) {
+      return Array.from(repoRefs);
+    }
+
+    // TODO(usergenic): When there are repos and tests with wildcards, we
+    // get two progress bars, identically labeled.  We should move the work to
+    // fetch the pages of repos into a support method that can be called in
+    // advance of the expand call and put the progress bar message there.
+    const allGitHubRepoRefs: git.GitHubRepoRef[] =
+        (await util.promiseAllWithProgress(
+             Array.from(ownersToFetchRepoNamesFor)
+                 .map(owner => this._github.getRepoFullNames(owner)),
+             'Fetching repo names for wildcard search'))
+            .reduce((a, b) => a.concat(b))
+            .map(git.parseGitHubRepoRefString);
+    const expandedRepoRefs: git.GitHubRepoRef[] = [];
+    for (const repoRef of repoRefs) {
+      if (repoRef.repoName.match(/\*/)) {
+        expandedRepoRefs.push.apply(
+            expandedRepoRefs,
+            allGitHubRepoRefs.filter(
+                otherRepoRef => git.matchRepoRef(repoRef, otherRepoRef)));
+      } else {
+        expandedRepoRefs.push(repoRef);
+      }
+    }
+    return expandedRepoRefs;
+  }
+
+  /**
+   * Cleans up the workspace folder and fixes repos which may be in
+   * incomplete
+   * or bad state due to prior abended runs.
```

Comment:
Is `abend` too archaic a reference "https://en.wikipedia.org/wiki/Abnormal_end"?